### PR TITLE
Github CI

### PR DESCRIPTION
**Pull request by hannesa2**
Before I included more changes, to have less warnings and errors.
But I guess it's better first to build current state and after this to improve it

### CODE REVIEW

**Comment by hannesa2** [comment]
Ok, now we see current state of master https://github.com/gitx/gitx/pull/208/checks?check_run_id=1767742541

image

**Comment by jspiro** [comment]
Screenshot 2021-01-26 at 12 20 32 AM
haha

File: .github/workflows/BuildPR.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: PullRequest
+
+on: [pull_request]
+
+jobs:
+  buildGitx:
```
Comment (jspiro) [inline .github/workflows/BuildPR.yml:6]:
```suggestion
  build-gitx:
```

File: .github/workflows/BuildPR.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+name: PullRequest
```
Comment (jspiro) [inline .github/workflows/BuildPR.yml:1]:
```suggestion
name: build-gitx
```

**Comment by jspiro** [comment]
You'll also want to update the readme with this badge after you rename the action, e.g.: https://github.com/gitx/gitx/workflows/build-gitx/badge.svg

File: .github/workflows/BuildPR.yml (hunk 2 above)
Comment (jspiro) [reply .github/workflows/BuildPR.yml]: Then https://github.com/gitx/gitx/workflows/build-gitx/badge.svg should work I think

**Comment by jspiro** [comment]
Pushed the README update. That's it for me for tonight, good luck and have a great day!

**Comment by hannesa2** [comment]
I changed this 
image


to get rid of 
image


Other wise we can only do pull request one-after-the-other or introduce ugly and pointless merge-master-to-branch-commits

**Comment by hannesa2** [comment]
How to solve this ?

image

I guess it's on our side, am I right ?

image

**Comment by jspiro** [comment]
```
$ find . -name verify.c
./External/objective-git/External/openssl/apps/verify.c
./External/Sparkle/Vendor/ed25519/src/verify.c
```

My guess is the submodules are not fully cloned, or that Sparkle removed/renamed this file in the SHA you moved the submodule to. Did we need to change Sparkle?

**Comment by jspiro** [comment]
```
$ rm -rf External/Sparkle
$ git submodule update --init --recursive
Submodule path 'External/Sparkle': checked out 'd5b0804b4a659ccb78c02acf56576fb08ce5d4d7'
Submodule path 'External/Sparkle/Vendor/ed25519': checked out '7fa6712ef5d581a6981ec2b08ee623314cd1d1c4'
$ find . -name verify.c
./External/objective-git/External/openssl/apps/verify.c
./External/Sparkle/Vendor/ed25519/src/verify.c
```

Seems like it's still there. So must be an incomplete clone.

**Comment by hannesa2** [comment]
I found the reason https://github.com/sparkle-project/Sparkle/tree/d5222353e6d6ee7baf4c18711e5b897fe2ed886a/Vendor `Sparkle` comes with this commit with own submodules , that's why we need a recursive submodule checkout

All other pull requests and improvements should now base on this !

**Comment by hannesa2** [comment]
I removed Travis because of build errors https://travis-ci.org/github/gitx/gitx/builds/756379732

File: README.markdown [hunk 3]
```diff
@@ -1,6 +1,7 @@
 # What is GitX?
 
-[![Build Status](https://travis-ci.org/gitx/gitx.svg?branch=master)](https://travis-ci.org/gitx/gitx)
+[![GitHub Actions Build Status](https://github.com/gitx/gitx/workflows/build-gitx/badge.svg)](https://github.com/gitx/gitx/actions?query=workflow%3Abuild-gitx)
+[![Travis Build Status](https://travis-ci.org/gitx/gitx.svg?branch=master)](https://travis-ci.org/gitx/gitx)
```
Comment (jspiro) [inline README.markdown:4]:
```suggestion
```

**Review by jspiro — APPROVED** [review]
Are there any other remnants of travis that we need to remove?

**Comment by hannesa2** [comment]
> Are there any other remnants of travis that we need to remove?

Cool, you found more. Currently no more are known to me

**Pull request merged**